File: source/bxgenbb_help/src/decay0/Sc48.cc

#include <sstream>
#include <stdexcept>
#include <cmath>

#include <mygsl/rng.h>

#include <genbb_help/decay0/Sc48.h>
#include <genbb_help/primary_event.h>
#include <genbb_help/decay0/alpha.h>
#include <genbb_help/decay0/gamma.h>
#include <genbb_help/decay0/electron.h>
#include <genbb_help/decay0/positron.h>
#include <genbb_help/decay0/particle.h>
#include <genbb_help/decay0/pair.h>
#include <genbb_help/decay0/nucltransK.h>
#include <genbb_help/decay0/nucltransKL.h>
#include <genbb_help/decay0/nucltransKLM.h>
#include <genbb_help/decay0/nucltransKLM_Pb.h>
#include <genbb_help/decay0/beta.h>
#include <genbb_help/decay0/beta1.h>
#include <genbb_help/decay0/beta2.h>
#include <genbb_help/decay0/beta_1fu.h>
#include <genbb_help/decay0/PbAtShell.h>

namespace genbb {
  namespace decay0 {

    // Sc48.f
    // This file was extracted from the 'decay0' program by V.I. Tretyak
    // Copyright 1995-2011 V.I. Tretyak
    // This program is free software
    // it under the terms of the GNU General Public License as published by
    // the Free Software Foundation
    // your option) any later version.
    //
    // This program is distributed in the hope that it will be useful, but
    // WITHOUT ANY WARRANTY
    // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
    // General Public License for more details.
    //
    // You should have received a copy of the GNU General Public License
    // along with this program
    // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
    //

    void Sc48(mygsl::rng & prng_, genbb::primary_event & event_, double tcnuc, double & tdnuc)
    {
      double t;
      double tdlev;
      double p;
      double pbeta;
      double tclev;
      double thlev;
      double thnuc;
      // Scheme of Sc48 decay ("Table of Isotopes", 8 ed., 1996 + NDS 68(1993)1).
      // Three-figured labels correspond to energies of 48Ti excited
      // levels in keV.
      // Input : tcnuc - time of creation of nucleus (sec)
      // Output: tdnuc - time of decay of nucleus (sec)
      // // common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),// ptime(100).
      // VIT, 7.05.1998
      thnuc=1.57212e5;
      tdnuc=tcnuc-thnuc/std::log(2.)*std::log(prng_());
      tclev=0.;
      pbeta=100.*prng_();
      if (pbeta <= 10.02) goto label_3509;
      goto label_3333;
    label_3509  :
      decay0_beta(prng_, event_, 0.483,22.,0.,0.,t);
      ////    label_35090  :
      thlev=1.4e-12;
      p=100.*prng_();
      if (p <= 24.14) goto label_35091;
      goto label_35092;
    label_35091  :
      decay0_nucltransK(prng_, event_, 1.213,0.005,8.8e-5,0.1e-4,tclev,thlev,tdlev);
      goto label_22960;
    label_35092  :
      decay0_nucltransK(prng_, event_, 0.175,0.005,4.5e-3,0.,tclev,thlev,tdlev);
      goto label_33330;
    label_3333  :
      decay0_beta(prng_, event_, 0.659,22.,0.,0.,t);
    label_33330  :
      thlev=221.e-15;
      decay0_nucltransK(prng_, event_, 1.038,0.005,1.1e-4,0.,tclev,thlev,tdlev);
      goto label_22960;
    label_22960  :
      thlev=0.762e-12;
      decay0_nucltransK(prng_, event_, 1.312,0.005,9.7e-5,0.1e-4,tclev,thlev,tdlev);
      goto label_98400;
    label_98400  :
      thlev=4.04e-12;
      decay0_nucltransK(prng_, event_, 0.984,0.005,1.3e-4,0.,tclev,thlev,tdlev);
      return;
    }
    // end of Sc48.f




  } // end of namespace decay0
} // end of namespace genbb

// end of Sc48.cc
// Local Variables: --
// mode: c++ --
// End: --

File: source/bxgenbb_help/src/decay0_fortran/Sc48.f
c Sc48.f 
c
c This file was extracted from the 'decay0' program by V.I. Tretyak
c
c Copyright 1995-2011 V.I. Tretyak
c
c This program is free software; you can redistribute it and/or modify
c it under the terms of the GNU General Public License as published by
c the Free Software Foundation; either version 3 of the License, or (at
c your option) any later version.
c 
c This program is distributed in the hope that it will be useful, but
c WITHOUT ANY WARRANTY; without even the implied warranty of
c MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
c General Public License for more details.
c 
c You should have received a copy of the GNU General Public License
c along with this program; if not, write to the Free Software
c Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
c 
c
	subroutine Sc48(tcnuc,tdnuc)
c Scheme of Sc48 decay ("Table of Isotopes", 8 ed., 1996 + NDS 68(1993)1).
c Three-figured labels correspond to energies of 48Ti excited
c levels in keV.
c Input : tcnuc - time of creation of nucleus (sec);
c Output: tdnuc - time of decay of nucleus (sec);
c         common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),ptime(100).
c VIT, 7.05.1998; 13.08.2007 update to NDS 107(2006)1747.
	thnuc=1.57212e5
	tdnuc=tcnuc-thnuc/alog(2.)*alog(rnd1(d))
	tclev=0.
	pbeta=100.*rnd1(d)
	if(pbeta.le.10.02) go to 3509
	                   go to 3333
c-----------------------------------------------------------------------
3509	call beta(0.483,22.,0.,0.,t)
35090	thlev=1.4e-12
	p=100.*rnd1(d)
	if(p.le.24.14) go to 35091
	               go to 35092
35091	call nucltransK(1.213,0.005,8.8e-5,0.1e-4,tclev,thlev,tdlev)
	go to 22960
35092	call nucltransK(0.175,0.005,4.5e-3,0.,tclev,thlev,tdlev)
	go to 33330
c-----------------------------------------------------------------------
3333	call beta(0.659,22.,0.,0.,t)
33330	thlev=221.e-15
	call nucltransK(1.038,0.005,1.1e-4,0.,tclev,thlev,tdlev)
	go to 22960
c-----------------------------------------------------------------------
22960	thlev=0.762e-12
	call nucltransK(1.312,0.005,9.7e-5,0.1e-4,tclev,thlev,tdlev)
	go to 98400
c-----------------------------------------------------------------------
98400	thlev=4.04e-12
	call nucltransK(0.984,0.005,1.3e-4,0.,tclev,thlev,tdlev)
	return
c-----------------------------------------------------------------------
	end
c
c end of Sc48.f
c
c Local Variables: --
c mode: fortran --
c End: --
